src/test/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActorTest.scala

```diff
@@ -314,163 +314,46 @@ class TaskLauncherActorTest extends AkkaUnitTest {
       val matchedTasks: MatchedInstanceOps = promise.future.futureValue
       matchedTasks.opsWithSource.foreach(_.accept())
 
-      val runningInstance = f.marathonInstance.copy(instanceId = scheduledInstance.instanceId)
+      val runningInstance = TestInstanceBuilder.newBuilder(f.app.id).addTaskRunning().instance
       val update = InstanceUpdated(runningInstance, Some(runningInstance.state), Seq.empty)
       Mockito.when(instanceTracker.instancesBySpecSync).thenReturn(InstanceTracker.InstancesBySpec.forInstances(runningInstance))
-      val counts = sendUpdate(launcherRef, update)
+      launcherRef ! update
+      val counts = (launcherRef ? TaskLauncherActor.GetCount).futureValue.asInstanceOf[QueuedInstanceInfo]
 
       assert(counts.finalInstanceCount == 1)
       assert(!counts.inProgress)
       assert(counts.instancesLeftToLaunch == 0)
       assert(counts.instancesLeftToLaunch == 0)
     }
 
-    "Expunged task is removed from counts" in new Fixture {
-      val update = TaskStatusUpdateTestHelper.finished(f.marathonInstance).wrapped
-      val expectedCounts = QueuedInstanceInfo(f.app, inProgress = false, 0, 0, Timestamp(0), Timestamp(0))
-
-      Mockito.when(instanceTracker.instancesBySpecSync).thenReturn(InstanceTracker.InstancesBySpec.forInstances(f.marathonInstance))
-
-      val launcherRef = createLauncherRef()
-      launcherRef ! RateLimiterActor.DelayUpdate(f.app, clock.now())
+    "Revive offers if instance deleted" in new Fixture {
```

Comment:
this class was greatly simplified in the feature branch so I had to adjust the tests a bit more...